Add workaround for Term::ReadKey >= 2.37

mschout opened the pull request:
As of Term::ReadKey v2.34, Pinto was unable to pull this module.  In
v2.37, the author added Term::ReadKey to the "provides" section of
META.yml.  This workaround fixes can_index() so that Pinto is able to
index Term::ReadKey 2.37 or later.

Fixes #240

mschout committed:
Add workaround for Term::ReadKey >= 2.37

As of Term::ReadKey v2.34, Pinto was unable to pull this module.  In
v2.37, the author added Term::ReadKey to the "provides" section of
META.yml.  This workaround fixes can_index() so that Pinto is able to
index Term::ReadKey 2.37 or later.

Fixes #240

thaljef commented:
Hi Michael, sorry for the long wait on this. The metadata for Term::ReadKey seems broken to me. I can't think of a reason why the metadata should say that `META.yml` is the file that provides the package. I'll open a ticket with the author to get an explanation.

mschout commented:
thanks

hartzell commented:
I left a comment over in jonathanstowe/TermReadKey#29 about this, I believe that the fix in the PR should be generalized for any case where the file is `META.{json,yaml}`.

@mschout -- are you up for that or should I swing at it?

mschout commented:
@hartzell I'll update my workaround today with your suggestions. Thanks

mschout committed:
Fix indexing of modules that declare a provides file of META.{json,yml}

This treats dists that provide a package file of META.yml or META.json
as simile's.  This META.yml or .json is perfectly valid per the CPAN
Meta Spec, but Pinto was unable to deal with it.

This fixes pulling Term::ReadKey >= v2.37, and probably other modules as
well.  As a result, the Term::ReadKey specific workaround has been
removed

Added test case for pulling a dist where the provides file is either
META.yml or META.json

Fixes #240
Fixes #241

mschout commented:
I removed the Term::ReadKey hack, and fixed it in is_simile as suggested.  I also added a test case.

mschout commented:
It appears Travis-CI is broken (fails to fetch Dist::Zilla). All tests pass on my machine

thaljef commented:
Merged. Thanks. I'll fix the Travis problems on the master branch.
